components/api-gateway-migrator/pkg/migrator/migrator.go

```diff
@@ -0,0 +1,297 @@
+	if m.skip() {
+		return m
+	}
+
+	if m.oldAPI.Labels != nil && m.oldAPI.Labels["migration/status"] != "" {
+		return m
+	}
+
+	log.Infof("disabling old api: %s/%s", m.oldAPI.Namespace, m.oldAPI.Name)
+
+	oldApiHost := m.oldAPI.Spec.Hostname
+	tempOldApiHost := generateTemporaryHost(m.oldAPI)
+
+	var updateFunc oldApiUpdateFunc = func(oldApi *oldapi.Api) {
+		oldApi.Spec.Hostname = tempOldApiHost
+		setOldApiLabel(oldApi, "migration/host", oldApiHost)
+		setOldApiLabel(oldApi, "migration/status", "disabled")
+	}
+
+	if err := m.updateOldApi(updateFunc); err != nil {
+		m.failure = err
+		return m
+	}
+
+	log.Infof("successfully disabled old api: %s/%s", m.oldAPI.Namespace, m.oldAPI.Name)
+	return m
+}
+
+// change the hostname of new apirule to the proper one
+func (m *Migrator) enableNewApi() *Migrator {
+	if m.skip() {
+		return m
+	}
+
+	log.Infof("enabling apirule for api: %s/%s", m.oldAPI.Namespace, m.oldAPI.Name)
+	host := m.oldAPI.Spec.Hostname
+
+	var updateFunc newApiUpdateFunc = func(newApi *newapi.APIRule) {
+		newApi.Spec.Service.Host = &host
+		setNewApiLabel(newApi, "migratedFrom", m.oldAPI.Name)
+	}
+
+	if err := m.updateNewApiRule(updateFunc); err != nil {
+		m.failure = err
+		// TODO: check if error is returned when nothing changed in apirule (host was already changed to proper value)
+		return m
+	}
+	log.Infof("successfully enabled apirule for api: %s/%s", m.oldAPI.Namespace, m.oldAPI.Name)
+	return m
+}
+
+// set label on an old api meaning that the object was migrated
+func (m *Migrator) markOldApiMigrated() *Migrator {
+	if m.skip() {
+		return m
+	}
+
+	if m.oldAPI.Labels != nil && m.oldAPI.Labels["migration/status"] == "migrated" {
+		return m
+	}
+
+	log.Infof("marking old api as migrated: %s/%s", m.oldAPI.Namespace, m.oldAPI.Name)
+
+	var updateFunc oldApiUpdateFunc = func(oldApi *oldapi.Api) {
+		oldApi.Labels["migration/status"] = "migrated"
+	}
+
+	if err := m.updateOldApi(updateFunc); err != nil {
+		m.failure = err
+		return m
+	}
+
+	log.Infof("successfully marked old api as migrated: %s/%s", m.oldAPI.Namespace, m.oldAPI.Name)
+	return m
+}
+
+func (m *Migrator) verifyNewApi() *Migrator {
+	if m.skip() {
+		return m
+	}
+
+	newApi, err := m.k8sClient.findTemporaryApiRule(m.oldAPI.Name)
+	if err != nil {
+		m.failure = err
+		return m
+	}
+	if newApi.Status.APIRuleStatus.Code != newapi.StatusOK {
+		m.failure = errors.New(fmt.Sprintf("Invalid ApiRule status: %s", newApi.Status.APIRuleStatus.Code))
+		return m
+	}
+	if newApi.Status.VirtualServiceStatus.Code != newapi.StatusOK {
+		m.failure = errors.New(fmt.Sprintf("Invalid VirtualService status: %s", newApi.Status.VirtualServiceStatus.Code))
+		return m
+	}
+	if newApi.Status.AccessRuleStatus.Code != newapi.StatusOK {
+		m.failure = errors.New(fmt.Sprintf("Invalid AccessRule status: %s", newApi.Status.AccessRuleStatus.Code))
+	}
+	return m
+}
+
+func generateApiRuleName(oldApi *oldapi.Api) string {
+	return oldApi.Name + "-" + generateRandomString(6)
```

Comment:
For security reasons consider adding truncating to 64 chars

Reply:
Done. At most 63 characters.